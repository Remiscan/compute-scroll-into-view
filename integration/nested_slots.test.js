beforeAll(async () => {
  await page.goto('http://localhost:3000/integration/nested_slots')
})

describe('target in nested slots into scrollable container in shadow DOM', () => {
  test('block: "nearest"', async () => {
    expect.assertions(4)
    const windowHeight = await page.evaluate(() => window.innerHeight)
    const actual = await page.evaluate(() => {
    return window
      .computeScrollIntoView(document.querySelector('.target'), {
        block: 'nearest',
      })
      .map(window.mapActions)
    })
    expect(actual).toHaveLength(3)
    expect(actual[0]).toMatchObject({ el: 'div.second-inner-container', left: 0, top: 100 })
    expect(actual[1]).toMatchObject({ el: 'div.first-inner-container', left: 0, top: windowHeight })
    expect(actual).toMatchSnapshot()
  })

  test('block: "start"', async () => {
    expect.assertions(4)
    const windowHeight = await page.evaluate(() => window.innerHeight)
    const actual = await page.evaluate(() => {
      return window
      .computeScrollIntoView(document.querySelector('.target'), {
        block: 'start',
      })
      .map(window.mapActions)
    })
    expect(actual).toHaveLength(3)
    expect(actual[0]).toMatchObject({ el: 'div.second-inner-container', left: 0, top: windowHeight })
    expect(actual[1]).toMatchObject({ el: 'div.first-inner-container', left: 0, top: windowHeight })
    expect(actual).toMatchSnapshot()
  })
})