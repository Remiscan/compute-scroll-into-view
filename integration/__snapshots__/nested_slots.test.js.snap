// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`target in nested slots into scrollable container in shadow DOM block: "nearest" 1`] = `
[
  {
    "el": "div.second-inner-container",
    "left": 0,
    "top": 100,
  },
  {
    "el": "div.first-inner-container",
    "left": 0,
    "top": 600,
  },
  {
    "el": "html",
    "left": 0,
    "top": 0,
  },
]
`;

exports[`target in nested slots into scrollable container in shadow DOM block: "start" 1`] = `
[
  {
    "el": "div.second-inner-container",
    "left": 0,
    "top": 600,
  },
  {
    "el": "div.first-inner-container",
    "left": 0,
    "top": 600,
  },
  {
    "el": "html",
    "left": 0,
    "top": 0,
  },
]
`;
